### PR TITLE
fix: do not allow to answer a poll after voting period ends

### PR DESCRIPTION
**Pull request by RiccardoM**
## Description

This PR fixes a bug within the `x/posts` module that currently allows users to answer a poll even when it's already ended.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [x] followed the guidelines for [building modules](https://docs.cosmos.network/v0.44/building-modules/intro.html)
- [x] included the necessary unit and integration [tests](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#testing)
- [x] added a changelog entry to `CHANGELOG.md`
- [x] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [x] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/posts/legacy/v4/store.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+// To do this, it iterates over all the polls, and removes from the store the user answers for
+// polls that are already ended (and thus should have not accepted new answers).
+func MigrateStore(ctx sdk.Context, storeKey sdk.StoreKey, cdc codec.BinaryCodec) error {
+	store := ctx.KVStore(storeKey)
+	return removeUserAnswers(store, cdc)
+}
+
+// removeUserAnswers iterates over all the polls and deletes the user answers for all the polls which
+// final results already been tallied. This is to delete all the answers that have been added after that.
+func removeUserAnswers(store sdk.KVStore, cdc codec.BinaryCodec) error {
```
Comment (dadamu) [inline x/posts/legacy/v4/store.go:21]: I would name it `removeInvalidAnswers`.

File: x/posts/legacy/v4/store.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+// removePollAnswers removes from the store the answers related to the given poll
+func removePollAnswers(store sdk.KVStore, subspaceID uint64, postID uint64, pollID uint32) error {
+	answersStore := prefix.NewStore(store, types.PollAnswersPrefix(subspaceID, postID, pollID))
+	answersIterator := answersStore.Iterator(nil, nil)
+
+	// Get the answers
+	var keys [][]byte
+	for ; answersIterator.Valid(); answersIterator.Next() {
+		keys = append(keys, append(types.PollAnswersPrefix(subspaceID, postID, pollID), answersIterator.Key()...))
```
Comment (dadamu) [inline x/posts/legacy/v4/store.go:55]:
```suggestion
		user := string(answersIterator.Key())
		keys = append(keys, types.PollAnswerStoreKey(subspaceID, postID, pollID, user))
```

File: x/posts/legacy/v4/store.go (hunk 1 above)
Comment (RiccardoM) [reply x/posts/legacy/v4/store.go]: Done!

**Review by dadamu — APPROVED** [review]
Great to me

**Pull request merged**